src/library/scala/collection/immutable/Queue.scala

```diff
@@ -118,7 +118,9 @@ sealed class Queue[+A] protected(protected val in: List[A], protected val out: L
     *
     *  @param  elem        the element to insert
     */
-  def enqueue[B >: A](elem: B): Queue[B] = new Queue(elem :: in, out)
+  def enqueue[B >: A](elem: B): Queue[B] =
+    if (isEmpty) new Queue(Nil, out :+ elem)
```

Comment:
We generally don't want to append things (`:+`) to `List`s - it is very slow. In this case however, we actually know the whole `Queue` is empty, so it can be `elem :: Nil`.